packages/flutter/lib/src/widgets/router.dart

```diff
@@ -1311,6 +1314,23 @@ abstract class RouteInformationProvider extends ValueListenable<RouteInformation
   /// The [routeInformation] is the new route information after the navigation
   /// event.
   void routerReportsNewRouteInformation(RouteInformation routeInformation) {}
+
+  /// A callback called when the [Router] widget detects [RouteInformation]
```

Comment:
Do we need separate API for this? From the docs it's not super clear when routerReportsNewRouteInformation is called over this one of vice versa.

Reply:
They were both update method when router detect routeinformation changes. The difference between them is whether this is a navigation event or not. Alternatively, I can have a flag to denote whether this is a navigation event or not, but it may be a bigger breaking change because this method will be called more often than before.

Reply:
Although bigger is relative to current one, I don't think a lot of people that wrote their own route information provider

Reply:
Depending on how breaking it is, a flag may actually be the nicer API. The flag would then indicate whether the new route information are supposed to replace the existing ones or whether they should be treated as a new entry.